src/mapgen/mapgen.cpp

```diff
@@ -215,6 +215,21 @@ void Mapgen::getMapgenNames(std::vector<const char *> *mgnames, bool include_hid
 	}
 }
 
+void Mapgen::setDefaultSettings(Settings *settings)
+{
+	// Option 1: List all defaults here as an overview
+	// Option 2: Each mapgen contains its own default setter function
+	// This is option 2.
+
+	settings->setDefault("mg_flags", flagdesc_mapgen,
+		 MG_CAVES | MG_DUNGEONS | MG_LIGHT | MG_DECORATIONS | MG_BIOMES);
```

Comment:
Does this avoid forcing dungeons to enabled for old worlds that had them disabled? Just checking as this default is different from before.

Reply:
This is exactly why I posted an example about the map compatibility in my last post. If it's not specified, it will be read as "not set" without any defaulting mechanisms that happen in `minetest.conf`.

Feel free to create a world, then remove `dungeons` from the flags to see what happens (becomes `nodungeons`).

Reply:
Ok, no problem.